x-pack/plugins/rule_registry/server/rule_data_client/index.ts

```diff
@@ -96,7 +96,8 @@ export class RuleDataClient implements IRuleDataClient {
           if (response.body.errors) {
             if (
               response.body.items.length > 0 &&
-              response.body.items?.[0]?.index?.error?.type === 'index_not_found_exception'
+              (response.body.items?.[0]?.index?.error?.type === 'index_not_found_exception' ||
+                response.body.items?.[0]?.index?.error?.type === 'illegal_argument_exception')
```

Comment:
`illegal_argument_exception` is returned if the alias exists but has no write_index.

Reply:
That's a good point, I'll add logic similar to the old "alias exists" check in `createWriteTargetIfNeeded` but instead checking to ensure the alias has no write target before attempting to create a write target for it. You're right that otherwise we could end up with `illegal_argument_exceptions` that cause us to try creating a new write target and that would cause problems if the alias already has a write target.

Reply:
I replaced the `aliasExists` check with a more general "index exists" check on the AAD indices. This makes the `createWriteTargetIfNeeded` function safer to call optimistically while still allowing the security solution to add AAD aliases to our `.siem-signals` indices. Now, if we do get an `illegal_argument_exception` for a reason other than "no concrete AAD indices exist", then `createWriteTargetIfNeeded` will return without making new indices and potentially making things worse.